base/MAPL_SwathGridFactory.F90

```diff
@@ -0,0 +1,1405 @@
+      _RETURN(_SUCCESS)
+
+   end function make_new_grid
+
+
+   function create_basic_grid(this, unusable, rc) result(grid)
+      type (ESMF_Grid) :: grid
+      class (SwathGridFactory), intent(in) :: this
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: status
+
+      _UNUSED_DUMMY(unusable)
+      
+      grid = ESMF_GridCreateNoPeriDim( &
+           & name = this%grid_name, &
+           & countsPerDEDim1=this%ims, &
+           & countsPerDEDim2=this%jms, &
+           & indexFlag=ESMF_INDEX_DELOCAL, &
+           & coordDep1=[1,2], &
+           & coordDep2=[1,2], &
+           & coordSys=ESMF_COORDSYS_SPH_RAD, &
+           & _RC)
+      
+      ! Allocate coords at default stagger location
+      call ESMF_GridAddCoord(grid, _RC)
+      call ESMF_GridAddCoord(grid, staggerloc=ESMF_STAGGERLOC_CORNER, _RC)
+
+      if (this%lm /= MAPL_UNDEFINED_INTEGER) then
+         call ESMF_AttributeSet(grid, name='GRID_LM', value=this%lm, _RC)
+      end if
+
+      call ESMF_AttributeSet(grid, 'GridType', 'LatLon', _RC)   ! grid=ESMF_grid
+
+      call ESMF_AttributeSet(grid, 'Global', .false., rc=status)
+
+
+      _RETURN(_SUCCESS)
+   end function create_basic_grid
+
+   
+
+   subroutine add_horz_coordinates_from_file(this, grid, unusable, rc)
+      use MAPL_BaseMod, only: MAPL_grid_interior
+      class (SwathGridFactory), intent(in) :: this
+      type (ESMF_Grid), intent(inout) :: grid
+      class (KeywordEnforcer), optional, intent(in) :: unusable
+      integer, optional, intent(out) :: rc
+
+      integer :: i_1, i_n, j_1, j_n ! regional array bounds
+      integer :: ic_1,ic_n,jc_1,jc_n ! regional corner bounds
+
+      !! shared mem
+      real(kind=ESMF_KIND_R8), pointer :: fptr(:,:)
+      real, pointer :: centers(:,:)               ! check R4 or R8
+      real, allocatable :: centers_full(:,:)
+      
+      integer :: status
+      integer :: i, j, ij(4), k
+      integer :: Xdim, Ydim
+      integer :: Xdim_full, Ydim_full
+      
+      integer :: IM, JM
+      integer :: IM_WORLD, JM_WORLD
+      integer :: COUNTS(3), DIMS(3)
+      character(len=:), allocatable :: lon_center_name, lat_center_name
+      
+      
+      _UNUSED_DUMMY(unusable)
+
+      ! keywords in netCDF
+      lon_center_name = "clon"
+      lat_center_name = "clat"
+      
+      call MAPL_grid_interior(grid, i_1, i_n, j_1, j_n)
+
+      !- shared mem case in MPI
+      !
+      Xdim=this%im_world
+      Ydim=this%jm_world
+
+      Xdim_full=this%cell_across_swath
+      Ydim_full=this%cell_along_swath
+
+
+      call MAPL_AllocateShared(centers,[Xdim,Ydim],transroot=.true.,_RC)
+      call MAPL_SyncSharedMemory(_RC)
+
+      write(6,*) 'Xdim, Ydim', Xdim, Ydim
+      write(6,*) 'Xdim_full, Ydim_full', Xdim_full, Ydim_full      
+      write(6,*) 'bf get_v2d_netcdf'
+      write(6,*) 'this%epoch_index(1:4)', this%epoch_index(1:4)
```

Comment:
These might be useful diagnostics for a while.   Please work with @mathomp4 to use pFlogger for this.